### PR TITLE
Update test framework featuregates type

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
A cleanup following #53025 and #57962.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
ref: #53025
and #57962.

**Special notes for your reviewer**:
but yeah, not sure if it's worthy to do this :)

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e_node/services/kubelet.go [hunk 1]
```diff
@@ -205,8 +206,9 @@ func (e *E2EServices) startKubelet() (*server, error) {
 
 	// Apply test framework feature gates by default. This could also be overridden
 	// by kubelet-flags.
-	if framework.TestContext.FeatureGates != "" {
-		cmdArgs = append(cmdArgs, "--feature-gates", framework.TestContext.FeatureGates)
+	if len(framework.TestContext.FeatureGates) != 0 {
+		featureGates := &utilflag.MapStringBool{Map: &framework.TestContext.FeatureGates}
+		cmdArgs = append(cmdArgs, "--feature-gates", featureGates.String())
```
Comment (mtaufen) [inline test/e2e_node/services/kubelet.go]: Why not 
```
cmdArgs = append(cmdArgs, "--feature-gates",
	         utilflag.NewMapStringBool(&framework.TestContext.FeatureGates).String())
```

Comment (xiangpengzhao) [reply test/e2e_node/services/kubelet.go]: Cool! Modified. Thanks!

**Comment by xiangpengzhao** [comment]
@mtaufen rebased and addressed inline comment. PTAL. Thanks!

File: test/e2e_node/services/kubelet.go [hunk 2]
```diff
@@ -266,9 +266,9 @@ func (e *E2EServices) startKubelet() (*server, error) {
 
 	// Apply test framework feature gates by default. This could also be overridden
 	// by kubelet-flags.
-	if framework.TestContext.FeatureGates != "" {
-		cmdArgs = append(cmdArgs, "--feature-gates", framework.TestContext.FeatureGates)
-		utilflag.NewMapStringBool(&kc.FeatureGates).Set(framework.TestContext.FeatureGates)
+	if len(framework.TestContext.FeatureGates) != 0 {
```
Comment (mtaufen) [inline test/e2e_node/services/kubelet.go]: prefer `if len(x) > 0`
semantically equivalent in this case but I think this is a more common style

**Review by mtaufen — COMMENTED** [review]
one more nit, then lgtm

File: test/e2e_node/services/kubelet.go (hunk 2 above)
Comment (xiangpengzhao) [reply test/e2e_node/services/kubelet.go]: done :)

**Comment by xiangpengzhao** [comment]
@mtaufen inline comment addressed. PTAL. Thanks!

**Comment by xiangpengzhao** [comment]
/retest
/assign @janetkuo
for approval. thanks!

**Pull request merged**